### DISCUSSION TITLE
Adding custom icons

### DISCUSSION THREAD

**Discussion by CoachKidd** [eth-p/obsidian-callout-manager#13] [Callout Manager Q&A]
Hi ... 

Is it possible to add custom icons to be used in callouts?
If so, how would I do it?

Thanks in advance?

**Comment by garugc** [comment]
I would also like to see more icons. I thought the plugin uses the icons in Obsidian. I would like to add an icon for yoga and can't find a suitable one for any of the following words: Yoga, Fitness, Exercise, Stretching

**Reply by eth-p** [reply]
Callout Manager uses the icons provided by Obsidian, yes.

Other plugins such as [Additional Icons](https://github.com/matthewturk/obsidian-additional-icons) add more icons to Obsidian, and Callout Manager is able to pick up on those. Unfortunately, I'm not aware of any plugin that both allows custom icons to be added, and that properly [adds them to the Obsidian icon list](https://docs.obsidian.md/Reference/TypeScript+API/addIcon).

For example, these icons were added by the above plugin:

<img width="1111" alt="image" src="https://github.com/eth-p/obsidian-callout-manager/assets/32112321/bea6c016-dce3-4ec8-949c-0d208f34c768">

**Comment by eth-p** [comment]
Hi @CoachKidd,

It is possible to add additional icons to Callout Manager, but I'm unaware of any plugin that allows you to add custom icons.

Callout Manager uses the [`getIconIds`](https://docs.obsidian.md/Reference/TypeScript+API/getIconIds#getIconIds()+function) function in Obsidian, since only those icons are available to be used in callouts. Any plugin that adds an icon using the [`addIcon`](https://docs.obsidian.md/Reference/TypeScript+API/addIcon) function should work with Callout Manager, and here's one that I've verified works:

- [Additional Icons](https://github.com/matthewturk/obsidian-additional-icons)
